### PR TITLE
Add a temporary hack to work around issue 88.

### PR DESCRIPTION
**Pull request by ndjc**
Can anyone else reproduce the issue with dcterms?
Can anyone suggest a better fix?

### CODE REVIEW

File: tools/ShapeChecker/src/main/java/net/open_services/scheck/shapechecker/HttpHandler.java [hunk 1]
```diff
@@ -39,11 +43,12 @@
      */
     public HttpHandler()
     {
-         httpClient = HttpClientBuilder
+        Header header = new BasicHeader(HttpHeaders.ACCEPT, "text/html;text/*;*/*");
```
Comment (berezovskyi) [inline tools/ShapeChecker/src/main/java/net/open_services/scheck/shapechecker/HttpHandler.java]: Shouldn’t this be a list of RDF MIME types?

File: tools/ShapeChecker/src/main/java/net/open_services/scheck/shapechecker/HttpHandler.java [hunk 2]
```diff
@@ -143,9 +150,32 @@ else if (containsMatch(skipURIPatterns,httpUri.toString()))
...
+    /**
+     * Hack to work around issue with Jena handling of Dublin Core redirects.
+     * @param httpUriOrig a URI
+     * @return if the input was a DCTerms URI, return a modified URI reflecting the several redirects
+     */
+    private static URI issue88(URI httpUriOrig)
+    {
+        Pattern dcterms = Pattern.compile("^http://purl\\.org/dc/terms/(.*)$");
+        URI httpUri = httpUriOrig;
+        Matcher m = dcterms.matcher(httpUri.toString());
```
Comment (berezovskyi) [inline tools/ShapeChecker/src/main/java/net/open_services/scheck/shapechecker/HttpHandler.java]: I think this match can be simply done with the startsWith method

File: tools/ShapeChecker/src/main/java/net/open_services/scheck/shapechecker/HttpHandler.java [hunk 3]
```diff
@@ -143,9 +150,32 @@ else if (containsMatch(skipURIPatterns,httpUri.toString()))
     }
 
 
+    /**
+     * Hack to work around issue with Jena handling of Dublin Core redirects.
+     * @param httpUriOrig a URI
+     * @return if the input was a DCTerms URI, return a modified URI reflecting the several redirects
+     */
+    private static URI issue88(URI httpUriOrig)
```
Comment (berezovskyi) [inline tools/ShapeChecker/src/main/java/net/open_services/scheck/shapechecker/HttpHandler.java]: Maybe call it dctermsRedirectWorkaround? You can link the issue in the javadoc.

File: tools/ShapeChecker/src/main/java/net/open_services/scheck/shapechecker/HttpHandler.java (hunk 1 above)
Comment (ndjc) [reply tools/ShapeChecker/src/main/java/net/open_services/scheck/shapechecker/HttpHandler.java]: No, these are the headers to be used when trying to fetch a resource that is not, or might not be, RDF. It turns out that some sites (including jazz.net) do not treat the absence of an ACCEPT header the same way as an ACCEPT header of "*/*", and then fail to redirect to the underlying real page. This was breaking some seeAlso links in OSLC vocabulary checks. Most seeAlso links are to HTML reference pages, hence the preference to get the HTML content type, if available.

File: tools/ShapeChecker/src/main/java/net/open_services/scheck/shapechecker/HttpHandler.java (hunk 2 above)
Comment (ndjc) [reply tools/ShapeChecker/src/main/java/net/open_services/scheck/shapechecker/HttpHandler.java]: The startsWith method doesn't capture the string after the starting match, and I need that to fill in the rest of the URI.

File: tools/ShapeChecker/src/main/java/net/open_services/scheck/shapechecker/HttpHandler.java (hunk 3 above)
Comment (ndjc) [reply tools/ShapeChecker/src/main/java/net/open_services/scheck/shapechecker/HttpHandler.java]: OK - I'll change this.

**Comment by berezovskyi** [comment]
@ndjc should we merge this?

**Pull request merged**